### PR TITLE
feat: allow airplane mode control

### PR DESCRIPTION
**Pull request by NyCodeGHG**
## Proposed Changes

add a new command for controlling the airplane mode of android devices.

## Testing

ran the following flow on android 27 and 28 emulators:
```yaml
appId: org.wikipedia
---
- launchApp
- setAirplaneMode: disable
- setAirplaneMode: enable
```

### CODE REVIEW

**Comment by axelniklasson** [comment]
@NyCodeGHG Nice 🎉 Let's update it to be a single command `- toggleAirplaneMode` instead and then we'll move forward with a formal review to get it merged. Thanks!

**Review by axelniklasson — CHANGES_REQUESTED** [review]
https://github.com/mobile-dev-inc/maestro/pull/1672#issuecomment-2125030154

**Comment by NyCodeGHG** [comment]
@axelniklasson hi, thank you for the review.
When I implemented this feature I had another option to toggle the airplane mode, but I removed it as it doesn't explicitly define the desired behavior. 
I can add another option to toggle if you want, but I'd prefer to keep the enable and disable options for those who want to be explicit.

**Comment by axelniklasson** [comment]
> @axelniklasson hi, thank you for the review. When I implemented this feature I had another option to toggle the airplane mode, but I removed it as it doesn't explicitly define the desired behavior. I can add another option to toggle if you want, but I'd prefer to keep the enable and disable options for those who want to be explicit.

Hey, that's fair and a good use case to me. Let's keep the API as it is but also add `- toggleAirplaneMode` as a separate command

File: maestro-orchestra-models/src/main/java/maestro/orchestra/Commands.kt [hunk 1]
```diff
@@ -866,6 +867,28 @@ data class StopRecordingCommand(
     }
 }
 
+enum class AirplaneValue {
+    @JsonProperty("enable")
```
Comment (axelniklasson) [inline maestro-orchestra-models/src/main/java/maestro/orchestra/Commands.kt:871]:
```suggestion
    @JsonProperty("enabled")
```

File: maestro-orchestra-models/src/main/java/maestro/orchestra/Commands.kt [hunk 2]
```diff
@@ -866,6 +867,28 @@ data class StopRecordingCommand(
...
+    Enable,
+    @JsonProperty("disable")
```
Comment (axelniklasson) [inline maestro-orchestra-models/src/main/java/maestro/orchestra/Commands.kt:873]:
```suggestion
    @JsonProperty("disabled")
```

File: maestro-orchestra-models/src/main/java/maestro/orchestra/Commands.kt [hunk 3]
```diff
@@ -866,6 +867,38 @@ data class StopRecordingCommand(
...
+    Disable,
+}
+
+data class SetAirplaneModeCommand(
+    val value: AirplaneValue,
+) : Command {
+    override fun description(): String {
+        return when (value) {
+            AirplaneValue.Enable -> "Enable Airplane mode"
+            AirplaneValue.Disable -> "Disable Airplane mode"
```
Comment (axelniklasson) [inline maestro-orchestra-models/src/main/java/maestro/orchestra/Commands.kt:883]:
```suggestion
            AirplaneValue.Enable -> "Enable airplane mode"
            AirplaneValue.Disable -> "Disable airplane mode"
```

**Review by axelniklasson — APPROVED** [review]
Nice, this works well for me! Can you also raise a PR in https://github.com/mobile-dev-inc/maestro-docs as well as adding another test case to the integration tests? Thanks!

**Comment by NyCodeGHG** [comment]
@axelniklasson 
I'm not sure on the integration test case
Can you elaborate?

**Comment by axelniklasson** [comment]
Thanks @NyCodeGHG! Merging this now but please do also raise a PR in https://github.com/mobile-dev-inc/maestro-docs documenting the new commands - we'll merge it when we make the next release 👍 

Thanks again for your contribution!

**Comment by NyCodeGHG** [comment]
will do!

**Pull request merged**